File: CPP/program/Car.cpp
#pragma once
// Baris ini memastikan bahwa file header hanya disertakan sekali dalam unit kompilasi.

#include <iostream> // Library ini digunakan untuk operasi input/output.
#include <string>   // Library ini digunakan untuk menggunakan tipe data string.
#include "Vehicle.cpp" // Ini adalah file header untuk kelas Vehicle.

using namespace std; // Baris ini menggunakan namespace standar.

// Ini adalah deklarasi dari kelas Car yang merupakan turunan dari kelas Vehicle.
class Car : public Vehicle{
    private:
        // Ini adalah anggota data pribadi dari kelas Car.
        string jenis_Car;
        int Jumlah_Kursi, Jumlah_Pintu;

    public:
        // Ini adalah konstruktor default dari kelas Car.
        Car(){
            Jumlah_Kursi = 0;
            Jumlah_Pintu = 0;
        }

        // Ini adalah konstruktor parameter dari kelas Car.
        Car(int Jumlah_Kursi,int Jumlah_Pintu){
            this->Jumlah_Kursi = Jumlah_Kursi;
            this->Jumlah_Pintu = Jumlah_Pintu;
        }

        // Ini adalah metode setter untuk mengatur nilai anggota data pribadi.
        void setJenisCar(string jenis_car){
            this->jenis_Car = jenis_car;
        }

        void setJumlahKursi(int Jumlah_Kursi) {
            this->Jumlah_Kursi = Jumlah_Kursi;
        }

        void setJumlahPintu(int Jumlah_Pintu) {
            this->Jumlah_Pintu = Jumlah_Pintu;
        }

        // Ini adalah metode getter untuk mendapatkan nilai anggota data pribadi.
        string getJenisCar(){
            return jenis_Car;
        }

        int getJumlahKursi() {
            return Jumlah_Kursi;
        }

        int getJumlahPintu() {
            return Jumlah_Pintu;
        }

        // Ini adalah destruktor dari kelas Car.
        ~Car(){
            
        }
};

File: Python/program/Car.py
# Ini adalah import kelas Vehicle yang diperlukan.
from Vehicle import Vehicle

# Ini adalah deklarasi dari kelas Car yang merupakan turunan dari kelas Vehicle.
class Car(Vehicle):
    def __init__(self, jumlah_kursi=0, jumlah_pintu=0):
        # Ini adalah konstruktor kelas Car.
        super().__init__()  # Memanggil konstruktor kelas induk (Vehicle)
        self.jenis_car = "" # Inisialisasi jenis_car dengan string kosong
        self.jumlah_kursi = jumlah_kursi
        self.jumlah_pintu = jumlah_pintu

    # Ini adalah metode setter untuk mengatur nilai atribut objek.
    def set_jenis_car(self, jenis_car):
        self.jenis_car = jenis_car

    def set_jumlah_kursi(self, jumlah_kursi):
        self.jumlah_kursi = jumlah_kursi

    def set_jumlah_pintu(self, jumlah_pintu):
        self.jumlah_pintu = jumlah_pintu

    # Ini adalah metode getter untuk mendapatkan nilai atribut objek.
    def get_jenis_car(self):
        return self.jenis_car

    def get_jumlah_kursi(self):
        return self.jumlah_kursi

    def get_jumlah_pintu(self):
        return self.jumlah_pintu

